Adjust wait_for_cooling slope

AnHardt opened the pull request:
Adjust wait_for_cooling slope
and drop mintemp for cooling.

See
https://github.com/MarlinFirmware/Marlin/pull/4169#issuecomment-231544532

thinkyhead commented on Marlin/Marlin_main.cpp:
```diff
@@ -4562,6 +4562,13 @@ inline void gcode_M105() {
 
 #endif
 
+  #if DISABLED(MIN_COOLING_SLOPE_DEG)
```
We have to use `#ifndef` for settings that take values.

thinkyhead replied:
We have to use `#ifndef` for settings that take values. So, once it's patched this up I'll merge this asap!

thinkyhead commented:
Overall it looks good. Do we need to increase one of our vars here to `long` as you mentioned?

AnHardt commented on Marlin/Marlin_main.cpp:
```diff
@@ -4769,11 +4783,11 @@ inline void gcode_M109() {
-          if (old_temp - temp < 1.0) break;
-          next_cool_check_ms = now + 20000;
+          if (old_temp - temp < MIN_COOLING_SLOPE_DEG_BED) break;
+          next_cool_check_ms = now + 1000UL * MIN_COOLING_SLOPE_TIME_BED;
```
@thinkyhead 
No, not a variable. Just a constant. For some reason `1000UL * MIN_COOLING_SLOPE_TIME_BED` is done in int if we don't  us UL. Worked great for times up to 32 seconds, but not for 60.

AnHardt committed:
Adjust wait_for_cooling slope

Adjust wait_for_cooling slope
and drop mintemp for cooling.

See
https://github.com/MarlinFirmware/Marlin/pull/4169#issuecomment-231544532
